python/libcuspatial/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# =============================================================================
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+cmake_minimum_required(VERSION 3.26.4 FATAL_ERROR)
+
+include(../../rapids_config.cmake)
+
+project(
+  libcuspatial-python
+  VERSION "${RAPIDS_VERSION}"
+  LANGUAGES CXX
+)
+
+# Check if cuspatial is already available. If so, it is the user's responsibility to ensure that the
+# CMake package is also available at build time of the Python cuspatial package.
+find_package(cuspatial "${RAPIDS_VERSION}")
+if(cuspatial_FOUND)
+  return()
+endif()
+
+unset(cuspatial_FOUND)
+
+# set options for building libcuspatial
+set(BUILD_SHARED_LIBS ON)
+set(BUILD_TESTS OFF)
+set(BUILD_BENCHMARKS OFF)
+set(CUDA_ENABLE_LINEINFO OFF)
+set(CUDA_STATIC_RUNTIME ON)
+set(CUSPATIAL_USE_CUDF_STATIC OFF)
+set(CUSPATIAL_EXCLUDE_CUDF_FROM_ALL ON)
+
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
```

Comment:
Could you remind me why we're setting this? Is it strictly necessary? Why aren't the install rules that come from `add_subdirectory` sufficient? If we do have to set this variable, why do we have to set it but also do a separate `install` below? It seems like either one or the other should be sufficient. I know this code is in the libcudf wheel now as well, but I don't understand why.

Reply:
I copied this in here from `libcudf` because I think I misunderstood the purpose there.

There, this is used to ensure all the nvcomp libraries and `libcudf.so` all get put together into the same place: https://github.com/rapidsai/cudf/blob/26091a44b3dbf0f56fc0dfc5f081077f2d00681f/python/libcudf/CMakeLists.txt#L50-L53

That isn't necessary here, and you're right that the install rules from the `add_subdirectory()` are sufficient. Removed this stuff in https://github.com/rapidsai/cuspatial/pull/1450/commits/fb307ba4ac1e0170e0e8d623b70cd913a4bbb077

Reply:
I suspect that the libcudf install rules are also a little overspecified now. Simultaneously setting both the CMake var and doing the install of cudf in particular seems redundant. The nvcomp installation might be fine as is. We can revisit those later, though.